### PR TITLE
Build as crate

### PR DESCRIPTION
**Pull request by sondrele**
A minimal example, that builds emlib as a stand-alone crate. I have not tried to build with cargo, but this is the next step.
- Copied the buttons_int example geckoboot.rs and all the files required to get this example to compile. The example can be found in `example/main.rs`.
- The library now only provides the modules `std`, `chip`, `cmu`, `gpio` and `emdrv`. The plan is to move `emdrv` to its own repository later, after we can build this project with cargo.
- The Makefiles have been modified slightly. It now creates two crates (core and emlib) and finds the main executable in `$(PROJ_DIR)/$(PROJ_NAME).rs`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+include Makefile.emlib
+include Makefile.rustlib
+
+
+LDFLAGS  = $(AFLAGS) -T$(LIB_PATH)/Device/SiliconLabs/EFM32GG/Source/GCC/efm32gg.ld
+LDFLAGS += -Wl,--start-group -lgcc -lc -lnosys -Wl,--end-group
+
+RUSTFLAGS  = --target $(TARGET)
+RUSTFLAGS += -g -C link-args="$(LDFLAGS)"
+RUSTFLAGS += -L . -L $(LIB_DIR) --verbose
```
Comment (havardh) [inline Makefile]: I'm guessing the -L . is not needed.

File: Makefile.rustlib [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+RUSTLIBFLAGS  = --crate-type lib --out-dir lib -O -g --target $(TARGET)
+RUSTLIBFLAGS += -L $(LIB_DIR) --cfg stage0
+
+$(LIB_DIR)/libcore.rlib: $(RUST_SRC_HOME)/libcore/lib.rs
+	$(RUSTC) $(RUSTLIBFLAGS) --crate-name core $<
+
+$(LIB_DIR)/libemlib.rlib: src/lib.rs
+	$(RUSTC) $(RUSTLIBFLAGS) --crate-name emlib $<
```
Comment (havardh) [inline Makefile.rustlib]: shouldn't --crate-name be specified in lib.rs?

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#![no_std]
+#![allow(unstable)]
+#![feature(lang_items)]
+
```
Comment (havardh) [inline src/lib.rs:4]: add 

```
#![crate_type="lib"]
#![crate_name="emlib"]
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+
+.PHONY: flash
+flash: all
+	$(FLASH) --flash $(OUT).bin $(FLASHFLAGS)
+
+.PHONY:clean
+clean:
+	@rm -rf lib
+	@rm -rf lib
```
Comment (havardh) [inline Makefile]: should one off these be `out`?

**Comment by sondrele** [comment]
I'm closing this pull-request, as it might soon get replaced with another one.

**Pull request closed without merging**